src/vs/workbench/browser/parts/editor/tabsTitleControl.ts

```diff
@@ -349,6 +349,18 @@ export class TabsTitleControl extends TitleControl {
 		tabContainer.setAttribute('role', 'presentation'); // cannot use role "tab" here due to https://github.com/Microsoft/vscode/issues/8659
 		DOM.addClass(tabContainer, 'tab monaco-editor-background');
 
+		if (!this.tabOptions.showTabCloseButton || this.tabOptions.tabCloseButton === 'off') {
```

Comment:
Please move this into the `doUpdate` method above where you will notice that there already is an update for `no-close-button` class from the other option (that should be deleted). We need to have it there because otherwise tabs will not update live from a config change.